src/compiler/checker.ts

```diff
@@ -7262,6 +7262,47 @@ namespace ts {
             }
         }
 
+        function isSuperCallExpression(n: Node): boolean {
+            return n.kind === SyntaxKind.CallExpression && (<CallExpression>n).expression.kind === SyntaxKind.SuperKeyword;
+        }
+
+        /**
+         * Return a cached result if super-statement is already found.
+         * Otherwise, find a super statement in a given constructor function and cache the result in the node-links of the constructor
+         *
+         * @param constructor constructor-function to look for super statement
+         */
+        function getSuperStatementInConstructor(constructor: ConstructorDeclaration): ExpressionStatement {
+            function getContainingSuperCall(n: Node): Node {

```

Comment:
`findFirstSuperCall` might be more appropriate. This certainly isn't the *containing* super call.

Also, move this out of the function, since it doesn't close over anything.